app/components/Inspector/HighlighterRect.js

```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import InspectorCSS from '../Inspector.css';
+
+/**
+ * Absolute positioned divs that overlay the app screenshot and highlight the bounding
+ * boxes of the elements in the app
+ */
+export default class HighlighterRect extends Component {
+
+  render () {
+    const {selectedElement = {}, selectHoveredElement, unselectHoveredElement, hoveredElement = {}, selectElement, unselectElement, node, zIndex, scaleRatio} = this.props;
+    const {path: hoveredPath} = hoveredElement;
+    const {path: selectedPath} = selectedElement;
+
+    let {bounds} = node.attributes || {};
+    if (bounds) {
+
+      // Parse the bounds from string to array
+      let [x1, y1, x2, y2] = bounds.split(/\[|\]|,/).filter((str) => str !== '');
+
+      // Calculate left, top, width and height coordinates
+      let left = x1 / scaleRatio;
+      let top = y1 / scaleRatio;
+      let width = (x2 - x1) / scaleRatio;
+      let height = (y2 - y1) / scaleRatio;
+
+      // Add class + special classes to hovered and selected elements
+      let highlighterClasses = [InspectorCSS['highlighter-box']];
+      hoveredPath === node.path && highlighterClasses.push(InspectorCSS['hovered-element-box']);
```

Comment:
I see. Yeah it's clever, but if the only point for the assignment is to get around syntax parsing rules to make this one-line if statement work, I'd rather remove it as it's confusing.